packages/core/src/model.ts

```diff
@@ -66,6 +67,21 @@ export function createModel(
         'context' in config ? config : { ...config, context: initialContext },
         implementations
       );
+    },
+    withInput: (input: any) => {
+      const model = createModel(initialContext, creators as any);
+
+      return {
+        ...model,
+        createMachine: (config, implementations) => {
+          return createMachine(
+            'context' in config
+              ? config
+              : { ...config, context: initialContext },
+            { input, ...implementations }
+          );
```

Comment:
that's equivalent to:
```suggestion
          return model.createMachine(config, { input, ...implementations });
```
which makes the behavior (regarding the `config.context`) of those two the same, out of the box

Reply:
actually - the order should be reversed here, `implementations` can already contain `input` and that shouldn't take precedence over the argument received by `withInput`:
```suggestion
          return model.createMachine(config, { ...implementations, input });
```